### PR TITLE
feat(ci): Cancel old jobs when new commits lands on PR

### PR DESCRIPTION
**Pull request by quentinDERORY**
Signed-off-by: quentinDERORY <15911421+quentinDERORY@users.noreply.github.com>

## Summary

Cancel old jobs when new commits lands on PR

## Test Plan

Test on PR

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by themarwhal — APPROVED** [review]

File: .github/workflows/agw-workflow.yml [hunk 1]
```diff
@@ -12,6 +12,10 @@ on:  # yamllint disable-line rule:truthy
       - v1.*
     types: [opened, reopened, synchronize]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
```
Comment (themarwhal) [inline .github/workflows/agw-workflow.yml:16]: what does the || mean in this context?

Comment (quentinDERORY) [reply .github/workflows/agw-workflow.yml:16]: it's the regular `or` .
The concurrency group will be the pr number if it's defined or the sha of the commit.

Comment (themarwhal) [reply .github/workflows/agw-workflow.yml:16]: ohhh i see - i guess pr number doesn't exist for master runs?

Comment (quentinDERORY) [reply .github/workflows/agw-workflow.yml:16]: yes exact!

**Review by hcgatewood — COMMENTED** [review]
This looks great! The one concern I see with this is we wouldn't get a green master build when there are successive merges to master -- so for especially long + flaky checks (lte-integ-test), this change might drastically reduce the number of passing commits

Is there any way to cancel-in-progress only if it's a pull request? cc @themarwhal

**Comment by quentinDERORY** [comment]
> This looks great! The one concern I see with this is we wouldn't get a green master build when there are successive merges to master -- so for especially long + flaky checks (lte-integ-test), this change might drastically reduce the number of passing commits
> 
> Is there any way to cancel-in-progress only if it's a pull request? cc @themarwhal

This is only cancelling for pull request. There will be no cancelling on master or on release branches

**Pull request merged**